lib/datastore.js

```diff
@@ -80,10 +80,37 @@ Datastore.prototype.loadDatabase = function () {
 
 
 /**
+ * Delete any documents from the array of documents that are expired
+ */
+Datastore.prototype._removeExpiredDocuments = function (docs) {
+  var self = this
+    , expiredDocs = []
+    , validDocs = [];
+
+  docs.forEach(function(doc) {
+    if (model.isExpired(doc) || doc.$$deleted === true) {
+      expiredDocs.push({ $$deleted: true, _id: doc._id });
+      self.removeFromIndexes(doc);
+    }
+    else {
+      validDocs.push(doc);
+    }
+  });
+
+  self.persistence.persistNewState(expiredDocs, function (err) {
+    //return callback(err || null);
```

Comment:
Disregard... I now allow an optional callback to be passed in to this function, though I see no immediate need to use it internally within NeDB.  I do have a use case for having it there as a consumer, though, for the sake of extending/overriding NeDB functions to support additional functionality (like emitting events, a la PR #156)... at least for now.